zwave_js_server/model/controller/firmware.py

```diff
@@ -0,0 +1,48 @@
+"""Provide a model for Z-Wave controller firmware."""
+from enum import IntEnum
+from typing import TypedDict
+
+
+class FirmwareUpdateStatus(IntEnum):
```

Comment:
I think it's confusing to have two classes with the same name. I suggest we name the controller and node firmware update status differently.

Reply:
I went back and forth on this. They are scoped to their import and represent the same thing in different use cases. We can add the extra info but in cases where they conflict the consumer should do that IMO

Reply:
I don't think it's a good idea. They don't have the same interface so they are not the same.

Same goes for the firmware update progress.